### DISCUSSION TITLE
Torrent is not added (rTorrent)

### DISCUSSION THREAD

**Discussion by TTR6214** [jesec/flood#86] [Q&A] [answered]
## Your Environment

* Version used:
  + Version : 4.0.2
  + Commit ID (development release) : Installed via npm flood@latest -g
* Environment name and version:
  + Node.js version v15.1.0
  + npm version 6.14.8
  + Web browser Chromium Version 73.0.3683.75 (Build de développement) built on Debian 9.8
* Operating System and version: running on Debian 9.13 (64 bits)

## Summary
I try to add a torrent via file (debian-10.6.0-amd64-netinst.iso.torrent) with Flood.
After click on "Add" button, I receive notification that the torrent is successfully added.
However the torrent is not added in rTorrent.

## Expected Behavior
The torrent should be added in rTorrent

## Current Behavior
The torrent is not added.
I can find it in the temp folder of Flood (/home/flood/.local/share/flood/temp/idNumber.torrent).
When I look at xmlrpc log of rtorrent, I can't see the file passing.
Flood is correctly set because I see my torrent list of rTorrent.
rTorrent is correctly set because I can add torrent with Transdrone/Transdroid (SCGI RPC).
I can also add torrent with Flood-UI from jfurrow.

## Possible Solution
Maybe it is permission issue because I use 2 users:
- flood user starts Flood UI service
- rtorrent user starts rTorrent daemon

**Comment by jesec** [comment, marked as answer]
I just did the "Add torrent by file" on my Flood instance. Unfortunately I can't reproduce the issue. 

Flood now passes the local file paths to rTorrent (96c754ddeb614b45a565e8307c9985ee85bcb7fa) in order bypass the XMLRPC limits to add larger files more reliably. 
(Flood-UI/flood#164, Flood-UI/flood#741, Flood-UI/flood#773)

You may have to make sure that the rTorrent can access files in Flood's temporary directory. Additionally, don't forget to make sure that the destination is allowed by Flood (see `--allowedpath` argument) and accessible by Flood.

**Comment by TTR6214** [comment]
Thank you for your quick answer, I follow your tip and solve my problem :+1: 
The folder /home/flood/.local/share was not accessible by user rtorrent (700 permission)
So I change folder permission to 755.
Now it's work!

However there was a bug, as Flood displayed that torrent is correctly added, but it was not.

**Comment by jesec** [comment]
Commit ac11f3190b13245e5b4609d4ce07b5368b3be971 might also solve the problem. I am not sure if rTorrent wants permissions on the parent directory, though. 

It is a known and long-standing problem that Flood doesn't handle errors reliably (and falsely treat errors as successes). 

https://github.com/jesec/flood/projects/6#card-44353690
https://github.com/Flood-UI/flood/issues/840

I will deal with the error handling problem eventually.

**Comment by TTR6214** [comment]
Actually, the folder and sub-folder of /home/flood/.local/share/flood was already 755 and the temp torrent file was 644
Only the folder share was 700, so I use chmod only for it (I didn't use the recursive option).

Thank you for your work and keep going ;)

**Comment by crainte** [comment]
Does this mean flood & rtorrent now have a requirement to be located on the same system?

**Comment by jesec** [comment]
> Does this mean flood & rtorrent now have a requirement to be located on the same system?

Flood **always** requires that, not just because of this behavior change. Many filesystem operations (filesystem browser, move torrents, delete torrent along with data, etc.) need the same context to work properly.

**Comment by xlanor** [comment]
fantastic work. This fork of flood has fixed the issue I had with my previous flood installation not adding torrents that were created with a larger chunk.
